pkg/render/common/components/components.go

```diff
@@ -171,6 +174,74 @@ func ApplyJobOverrides(job *batchv1.Job, overrides components.ReplicatedPodResou
 	job.Spec.Template = *r.podTemplateSpec
 }
 
+// ApplyStatefulSetOverrides applies the overrides to the given DaemonSet.
+// Note: overrides must not be nil pointer.
+func ApplyStatefulSetOverrides(s *appsv1.StatefulSet, overrides components.ReplicatedPodResourceOverrides) {
+	// Catch if caller passes in an explicit nil.
+	if overrides == nil {
+		return
+	}
+
+	// Pull out the data we'll override from the DaemonSet.
+	r := &replicatedPodResource{
+		labels:          s.Labels,
+		annotations:     s.Annotations,
+		minReadySeconds: &s.Spec.MinReadySeconds,
+		podTemplateSpec: &s.Spec.Template,
+	}
+	// Apply the overrides.
+	applyReplicatedPodResourceOverrides(r, overrides)
+
+	// Set the possibly new fields back onto the DaemonSet.
+	s.Labels = r.labels
+	s.Annotations = r.annotations
+	s.Spec.MinReadySeconds = *r.minReadySeconds
+	s.Spec.Template = *r.podTemplateSpec
+}
+
```

Comment:
Placed ApplyKibanaOverrides func inside components to take advantage of the existing replicated pod resources and applyReplicatedPodResourceOverrides func. Since kibana spec has Pod Template which can be overridden in the same way.